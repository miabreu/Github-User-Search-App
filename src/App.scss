* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

html, body {
  overflow-x: hidden;
  max-width: 100%;
}

body {
  // background: rgb(20, 29, 47);
  // color: #FFFFFF;
  font-family: 'Space Mono', monospace;
  font-size: 15px;
  line-height: 25px;

  h1 {
    font-size: 1.625rem;
    line-height: 38px;
  }

  h2 {
    font-size: 1.375rem;
    line-height: 33px;
  }

  h3 {
    font-size: 1rem;
    line-height: 24px;
  }

  h4 {
    font-size: 0.813rem;
    line-height: 20px;
  }
}

.container {
    display: flex;
    flex-direction: column;
    max-width: 730px;
    margin: 0 auto;
    padding: 2em 1.5em;
    min-height: 100vh;
    justify-content: center;

    .unavailable {
      color: rgba(246, 248, 255, 0.4);
    }

    .unavailable-img {
      filter: grayscale(25%);
    }
}

.dark {
  background: rgb(20, 29, 47);
  color: #FFFFFF;
  transition: background 0.8s;
  
  .profile-card {
    background: rgb(30, 42, 71);
    transition: background 0.8s;
  }

  .highlights {
    background: rgb(20, 29, 47);
    transition: background 0.8s;
  }

  .toggle-wrapper div:hover,
  .toggle-wrapper div:focus {
    filter: invert(53%) sepia(14%) saturate(1202%) hue-rotate(188deg) brightness(150%) contrast(80%);
  }
}

.light {
  background: rgb(246, 248, 255);
  color: rgb(43, 52, 66);
  transition: background 0.8s;

  .profile-card {
    background: rgb(254, 254, 254);
    box-shadow: 0 16px 30px -10px rgba(70, 96, 187, 19.86%);
    transition: background 0.8s;
  }

  .highlights {
    background: rgb(246, 248, 255);
    transition: background 0.8s;

    .highlight p {
      color: rgb(75, 106, 155);
    }
  }

  .location-social-info {
    p, a {
      color: rgb(75, 106, 155)
    }

    img {
      filter: brightness(1);
    }

    .unavailable {
      color: rgba(75, 106, 155, .5)
    }

    .unavailable-img {
      filter: brightness(1) grayscale(0.4) brightness(160%)
    }
  }

  .unavailable {
    color: rgba(75, 106, 155, .5)
  }

  .unavailable-img {
    filter: grayscale(10%);
  }

  .created-at {
    color: rgb(105, 124, 154);
  }

  .form-control {
    background: rgb(254, 254, 254);
    box-shadow: 0 16px 30px -10px rgba(70, 96, 187, 19.86%);
    transition: background 0.8s;

    .form-control-input {
      background: rgb(254, 254, 254);
      color: rgb(75, 106, 155);
      background: url('../src/assets/icon-search.svg') no-repeat 24px/24px;
      transition: background 0.8s;
    }

    .form-control-input::placeholder {
      color: rgb(75, 106, 155);
    }
  }

  .toggle-wrapper div {
    span {
      color: rgb(105, 124, 154);
    }
  }

  .toggle-wrapper div:hover,
  .toggle-wrapper div:focus {
    color: rgb(34, 39, 49);
    filter: brightness(0.5);
  }
}


@media (prefers-color-scheme: light) {
  // Light theme styles go here
}

@media (prefers-color-scheme: dark) {
  // Dark theme styles go here
}