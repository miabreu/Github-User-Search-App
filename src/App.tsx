import React from "react";
import Header from "./components/header/Header.component";
import Search from "./components/search/Search.component";
import ProfileCard from "./components/profilecard/ProfileCard.component";
import { UserData } from "./interfaces";

// Style imports
import './styles/base.scss';
import './styles/light-theme.scss';
import './styles/dark-theme.scss';

function App(): JSX.Element {
  const [userData, setUserData] = React.useState<UserData>({
      login: "Octocat",
      name: "",
      bio: "",
      created_at: new Date(),
      public_repos: 0,
      followers: 0,
      location: "",
      blog: "",
      company: "",
      twitter_username: "",
      following: 0, 
      avatar_url: "",
      url: ""
  });
  const [isDarkMode, setIsDarkMode] = React.useState<boolean>(true);
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const [isLoading, setIsLoading] = React.useState<boolean>(false);
  const [hasError, setHasError] = React.useState<boolean>(false);
  const [username, setUsername] = React.useState<string>("octocat");
  const searchInput = React.useRef<HTMLInputElement>(null);

  const handleToggle = () => {
    setIsDarkMode(darkMode => !darkMode);
  }

  React.useEffect(() => {
    const fetchInfo = async () => {
      setIsLoading(true);
      setHasError(false);
      try {
        const res = await fetch(`https://api.github.com/users/${username}`);
        // We want to check if the request was valid or not before continuing
        if(res.status !== 200) {
          throw new Error();
        }
        const json = await res.json();
        setUserData(json);
      } catch (error) {
        console.log(
          "Error in API request, please ensure the url is valid.",
        );
        setHasError(true);
      }
      setIsLoading(false);
    };
    fetchInfo();
  }, [username]);

  return (
    <main className={isDarkMode ? 'dark' : 'light'}>
      <div className="container">
        <Header 
          isDarkMode={isDarkMode} 
          toggleMode={handleToggle} 
        />
        <Search 
          updateUsername={setUsername} 
          searchInput={searchInput}
          hasError={hasError} />
        <ProfileCard user={userData} /> 
      </div>
    </main>
  );
}

export default App;
