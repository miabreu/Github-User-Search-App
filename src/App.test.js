import { render, cleanup } from '@testing-library/react';
import App from './App';
import Header from './components/header/Header.component.tsx';
import Toggle from './components/toggle/Toggle.component.tsx';
import ProfileCard from './components/profilecard/ProfileCard.component.tsx';
import Search from './components/search/Search.component.tsx';

// Clean out everything before running test again
afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<App />);

  expect(asFragment(<App />)).toMatchSnapshot();
});
test('Header component snapshot', () => {
  const { asFragment } = render(<Header />);

  expect(asFragment(<Header />)).toMatchSnapshot();
});

test('Toggle component snapshot', () => {
  const { asFragment } = render(<Toggle />);

  expect(asFragment(<Toggle />)).toMatchSnapshot();
});

test('Profile component snapshot', () => {
  const user = {};
  const { asFragment } = render(<ProfileCard user={user} />);

  expect(asFragment(<ProfileCard />)).toMatchSnapshot();
})

test('Search Component', () => {
  const { asFragment } = render(<Search />);

  expect(asFragment(<Search />)).toMatchSnapshot();
});
