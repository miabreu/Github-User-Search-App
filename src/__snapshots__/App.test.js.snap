// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component 1`] = `
<DocumentFragment>
  <header
    class="header-wrapper"
  >
    <h1
      class="title"
    >
      devfinder
    </h1>
    <div
      class="toggle-wrapper"
    >
      <div
        class="toggle"
      >
        <span>
          Dark
        </span>
        <img
          alt="Icon of a moon"
          src="icon-moon.svg"
        />
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Header component snapshot 1`] = `
<DocumentFragment>
  <header
    class="header-wrapper"
  >
    <h1
      class="title"
    >
      devfinder
    </h1>
    <div
      class="toggle-wrapper"
    >
      <div
        class="toggle"
      >
        <span>
          Dark
        </span>
        <img
          alt="Icon of a moon"
          src="icon-moon.svg"
        />
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Profile Component 1`] = `
<DocumentFragment>
  <div
    class="profile-card"
  >
    <div
      class="avatar-wrapper"
    />
    <div
      class="content"
    >
      <div
        class="header"
      >
        <div
          class="details"
        >
          <div
            class="title-wrapper"
          >
            <h2>
              Not found
            </h2>
            <a
              href=""
            />
          </div>
          <div
            class="created-at"
          >
            <p />
          </div>
        </div>
      </div>
      <div
        class="bio"
      >
        <p
          class="unavailable"
        >
           This profile has no bio.
        </p>
      </div>
      <div
        class="body"
      >
        <div
          class="highlights"
        >
          <div
            class="highlight"
          >
            <p>
              Repos
            </p>
            <span>
              0
            </span>
          </div>
          <div
            class="highlight"
          >
            <p>
              Followers
            </p>
            <span>
              0
            </span>
          </div>
          <div
            class="highlight"
          >
            <p>
              Following
            </p>
            <span>
              0
            </span>
          </div>
        </div>
        <div
          class="location-social-info"
        >
          <div
            class="location"
          >
            <img
              alt="Map marker icon to signal location"
              class="unavailable-img"
              src="icon-location.svg"
            />
            <p
              class="unavailable"
            >
              Not Available
            </p>
          </div>
          <div
            class="profile-link"
          >
            <img
              alt="Map marker icon to signal location"
              class="unavailable-img"
              src="icon-website.svg"
            />
            <p
              class="unavailable"
            >
              Not Available
            </p>
          </div>
          <div
            class="twitter-social"
          >
            <img
              alt="Map marker icon to signal location"
              class="unavailable-img"
              src="icon-twitter.svg"
            />
            <p
              class="unavailable"
            >
              Not Available
            </p>
          </div>
          <div
            class="company"
          >
            <img
              alt="Map marker icon to signal location"
              class="unavailable-img"
              src="icon-company.svg"
            />
            <p
              class="unavailable"
            >
              Not Available
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Profile component snapshot 1`] = `
<DocumentFragment>
  <div
    class="profile-card"
  >
    <div
      class="avatar-wrapper"
    />
    <div
      class="content"
    >
      <div
        class="header"
      >
        <div
          class="details"
        >
          <div
            class="title-wrapper"
          >
            <h2>
              Not found
            </h2>
            <a
              href=""
            />
          </div>
          <div
            class="created-at"
          >
            <p />
          </div>
        </div>
      </div>
      <div
        class="bio"
      >
        <p
          class="unavailable"
        >
           This profile has no bio.
        </p>
      </div>
      <div
        class="body"
      >
        <div
          class="highlights"
        >
          <div
            class="highlight"
          >
            <p>
              Repos
            </p>
            <span>
              0
            </span>
          </div>
          <div
            class="highlight"
          >
            <p>
              Followers
            </p>
            <span>
              0
            </span>
          </div>
          <div
            class="highlight"
          >
            <p>
              Following
            </p>
            <span>
              0
            </span>
          </div>
        </div>
        <div
          class="location-social-info"
        >
          <div
            class="location"
          >
            <img
              alt="Map marker icon to signal location"
              class="unavailable-img"
              src="icon-location.svg"
            />
            <p
              class="unavailable"
            >
              Not Available
            </p>
          </div>
          <div
            class="profile-link"
          >
            <img
              alt="Map marker icon to signal location"
              class="unavailable-img"
              src="icon-website.svg"
            />
            <p
              class="unavailable"
            >
              Not Available
            </p>
          </div>
          <div
            class="twitter-social"
          >
            <img
              alt="Map marker icon to signal location"
              class="unavailable-img"
              src="icon-twitter.svg"
            />
            <p
              class="unavailable"
            >
              Not Available
            </p>
          </div>
          <div
            class="company"
          >
            <img
              alt="Map marker icon to signal location"
              class="unavailable-img"
              src="icon-company.svg"
            />
            <p
              class="unavailable"
            >
              Not Available
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Search Component 1`] = `
<DocumentFragment>
  <div
    class="form-wrapper"
  >
    <form
      autocomplete="off"
      class="form-control"
    >
      <input
        class="form-control-input"
        name="searchTerm"
        placeholder="Search Github username..."
        type="text"
      />
      <input
        class="form-control-action"
        type="submit"
        value="Search"
      />
    </form>
  </div>
</DocumentFragment>
`;

exports[`Toggle Component 1`] = `
<DocumentFragment>
  <div
    class="toggle-wrapper"
  >
    <div
      class="toggle"
    >
      <span>
        Dark
      </span>
      <img
        alt="Icon of a moon"
        src="icon-moon.svg"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Toggle component snapshot 1`] = `
<DocumentFragment>
  <div
    class="toggle-wrapper"
  >
    <div
      class="toggle"
    >
      <span>
        Dark
      </span>
      <img
        alt="Icon of a moon"
        src="icon-moon.svg"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <main
    class="dark"
  >
    <div
      class="container"
    >
      <header
        class="header-wrapper"
      >
        <h1
          class="title"
        >
          devfinder
        </h1>
        <div
          class="toggle-wrapper"
        >
          <div
            class="toggle"
          >
            <span>
              Light
            </span>
            <img
              alt="Icon of a sun"
              src="icon-sun.svg"
            />
          </div>
        </div>
      </header>
      <div
        class="form-wrapper"
      >
        <form
          autocomplete="off"
          class="form-control"
        >
          <input
            class="form-control-input"
            name="searchTerm"
            placeholder="Search Github username..."
            type="text"
          />
          <input
            class="form-control-action"
            type="submit"
            value="Search"
          />
        </form>
      </div>
      <div
        class="profile-card"
      >
        <div
          class="avatar-wrapper"
        />
        <div
          class="content"
        >
          <div
            class="header"
          >
            <div
              class="details"
            >
              <div
                class="title-wrapper"
              >
                <h2>
                  Not found
                </h2>
                <a
                  href=""
                />
              </div>
              <div
                class="created-at"
              >
                <p>
                  Joined 14 Apr 2023
                </p>
              </div>
            </div>
          </div>
          <div
            class="bio"
          >
            <p
              class="unavailable"
            >
               This profile has no bio.
            </p>
          </div>
          <div
            class="body"
          >
            <div
              class="highlights"
            >
              <div
                class="highlight"
              >
                <p>
                  Repos
                </p>
                <span>
                  0
                </span>
              </div>
              <div
                class="highlight"
              >
                <p>
                  Followers
                </p>
                <span>
                  0
                </span>
              </div>
              <div
                class="highlight"
              >
                <p>
                  Following
                </p>
                <span>
                  0
                </span>
              </div>
            </div>
            <div
              class="location-social-info"
            >
              <div
                class="location"
              >
                <img
                  alt="Map marker icon to signal location"
                  class="unavailable-img"
                  src="icon-location.svg"
                />
                <p
                  class="unavailable"
                >
                  Not Available
                </p>
              </div>
              <div
                class="profile-link"
              >
                <img
                  alt="Map marker icon to signal location"
                  class="unavailable-img"
                  src="icon-website.svg"
                />
                <p
                  class="unavailable"
                >
                  Not Available
                </p>
              </div>
              <div
                class="twitter-social"
              >
                <img
                  alt="Map marker icon to signal location"
                  class="unavailable-img"
                  src="icon-twitter.svg"
                />
                <p
                  class="unavailable"
                >
                  Not Available
                </p>
              </div>
              <div
                class="company"
              >
                <img
                  alt="Map marker icon to signal location"
                  class="unavailable-img"
                  src="icon-company.svg"
                />
                <p
                  class="unavailable"
                >
                  Not Available
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
